crates/ignore/src/walk.rs

```diff
@@ -1682,15 +1681,10 @@ impl<'s> Worker<'s> {
                     return None;
                 }
                 None => {
-                    // Once num_pending reaches 0, it is impossible for it to
-                    // ever increase again. Namely, it only reaches 0 once
-                    // all jobs have run such that no jobs have produced more
-                    // work. We have this guarantee because num_pending is
-                    // always incremented before each job is submitted and only
-                    // decremented once each job is completely finished.
-                    // Therefore, if this reaches zero, then there can be no
-                    // other job running.
-                    if self.num_pending() == 0 {
+                    let active = self
+                        .active_workers
+                        .fetch_sub(1, AtomicOrdering::Relaxed);
+                    if active == 1 {
```

Comment:
Can you add a comment explaining the termination logic here?

Reply:
I updated the comment with a correctness argument in d6f850f57436301fea602a5b4419ffacb464bcc9